mw_strip_auth_test.go

```diff
@@ -0,0 +1,131 @@
+package main
+
+import (
+	"fmt"
+	"math/rand"
+	"net/http"
+	"testing"
+
+	"github.com/TykTechnologies/tyk/apidef"
+)
+
+const letters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+
+func randStringBytes(n int) string {
+	b := make([]byte, n)
+
+	for i := range b {
+		b[i] = letters[rand.Intn(len(letters))]
+	}
+
+	return string(b)
+}
+
+func TestStripAuth_Name(t *testing.T) {
+
+	sa := StripAuth{}
+
+	if sa.Name() != "StripAuth" {
+		t.Error("Incorrect Name")
+	}
+}
+
+func TestStripAuth_EnabledForSpec(t *testing.T) {
+
+	sa := StripAuth{}
+
+	sa.Spec = &APISpec{StripAuthData: true}
+
+	if sa.EnabledForSpec() != sa.Spec.StripAuthData {
+		t.Error("EnablesForSpec be true when sa.Spec.StripAuthData is true")
+	}
+
+	sa.Spec.StripAuthData = false
+
+	if sa.EnabledForSpec() != sa.Spec.StripAuthData {
+		t.Error("EnablesForSpec should false when sa.Spec.StripAuthData is false")
+	}
+}
```

Comment:
This one too - you're writing a dozen lines of test for a single line of trivial code.

If you want to be sure that the middleware works, don't unit test every single part of it. Unit test the complex bits like you do below, and then do a full end-to-end test. Should be easy with httptest and loading a custom spec.